src/main.rs

```diff
@@ -59,6 +70,112 @@ fn check_valid_env(
     Ok(())
 }
 
+fn current_env() -> HashMap<String, String> {
+    // Create a HashMap from the current set of environment variables (excluding a few)
+    let exclude = ["PS1", "TERM", "HOME"];
+
+    env::vars()
+        .filter(|&(ref k, _)| !exclude.contains(&k.as_str()))
```

Comment:
As far as I can tell, you don't need both `&` and `ref`.

Reply:
Thanks